PubNubUnity/Assets/PubNubUnity/PubNubUnityBase.cs

```diff
@@ -0,0 +1,125 @@
+        private const string build = "4.0-beta4";
+        private string pnsdkVersion = string.Format ("PubNub-CSharp-Unity/{0}", build);
+
+        public string Version {
+            get {
+                return pnsdkVersion;
+            }
+            private set{
+                pnsdkVersion = value;
+            }
+        }
+        
+        internal PNLoggingMethod PNLog;
+        public PNConfiguration PNConfig { get; set;}
+        internal QueueManager QManager { get; set;}
+        private IJsonLibrary jsonLibrary;
+        public IJsonLibrary JsonLibrary {
+            get {
+                if (jsonLibrary == null)
+                {
+                    jsonLibrary = JSONSerializer.JsonLibrary(this);
+                }
+                return jsonLibrary;
+            }
+
+            set {
+                if ((value != null) && (value is IJsonLibrary)) {
+                    jsonLibrary = value;
+                } else {
+                    jsonLibrary = JSONSerializer.JsonLibrary(this);
+                    #if (ENABLE_PUBNUB_LOGGING)
+                    this.PNLog.WriteToLog ("Missing or Incorrect JsonLibrary value, using default", PNLoggingMethod.LevelWarning);
+                    #endif
+                }
+            }
+        }
+        public GameObject GameObjectRef { get; set;}
+        public Subscription SubscriptionInstance { get; set;}
+        internal SubscriptionWorker<SubscribeEnvelope> SubWorker { get; set;}
+        internal bool localGobj;
+        public readonly PNLatencyManager latency;
+        public PNLatencyManager Latency{
+            get {return latency;}
+        }
+
+
+        public PubNubUnityBase(PNConfiguration pnConfiguration, GameObject gameObjectRef, IJsonLibrary jsonLibrary){
+            PNConfig = pnConfiguration;
+            PNLog = new PNLoggingMethod(PNConfig.LogVerbosity);
+            #if(UNITY_IOS)
+            Version = string.Format("PubNub-CSharp-UnityIOS/{0}", build);
+            #elif(UNITY_STANDALONE_WIN)
+            Version = string.Format("PubNub-CSharp-UnityWin/{0}", build);
+            #elif(UNITY_STANDALONE_OSX)
+            Version = string.Format("PubNub-CSharp-UnityOSX/{0}", build);
+            #elif(UNITY_ANDROID)
+            Version = string.Format("PubNub-CSharp-UnityAndroid/{0}", build);
+            #elif(UNITY_STANDALONE_LINUX)
+            Version = string.Format("PubNub-CSharp-UnityLinux/{0}", build);
+            #elif(UNITY_WEBPLAYER)
+            Version = string.Format("PubNub-CSharp-UnityWeb/{0}", build);
+            #elif(UNITY_WEBGL)
+            Version = string.Format("PubNub-CSharp-UnityWebGL/{0}", build);
+            #else
+            Version = string.Format("PubNub-CSharp-Unity/{0}", build);
+            #endif
+            #if (ENABLE_PUBNUB_LOGGING)
+            this.PNLog.WriteToLog (Version, PNLoggingMethod.LevelInfo);
+            #endif
+
+            if (GameObjectRef == null) {
```

Comment:
I'm guessing this is supposed to be gameObjectRef and that GameObjectRef should be assigned in the else-clause?

Reply:
thats also true, modified. Thanks @schakkis !